contents/tutorials/css-selectors-for-actions.md

```diff
@@ -0,0 +1,171 @@
+
+```html
+    <input
+        type='text'
+        name='text-input'
+    />
+```
+
+Attribute selectors in PostHog can match both the fact of "presence" of atrribute on the element and it's value. As of now PostHog does not support other attribute operators like not equal `~=`, partial match `^=`, etc. 
+
+> It's also worth noting that attribute selectors match all HTML tags with a given attribute disregarding the type of the tag. 
+
+*Example*: `[type]` selector will match all elements having `type` attribute (both empty and with some value). `[type='text']` selector will match only elements with `type='text'` (like a code snippet above).
+
+### Combinators
+
+Combinators allow to chain different selectors together.
+
+|Combinator Name|Syntax|Example|
+|---------------|------|-----|
+|Descendant combinator|`A B`|`div span` combinator wiil match `<span>` inside `<div>`|
+|Child combinator|`A > B`|`div > span` combinator wiil match `<span>` which is a direct child of `<div>`|
+|Sibling combinator|`A ~ B`|`div ~ span` combinator wiil match `<span>` following `<div>` immediately or not|
+|Adjacent sibling combinator|`A + B`|`div + span` combinator wiil match `<span>` immediately following `<div>`|
+
+**Selectors for pseudo-classes and pseudo-elements**
+These selectos are especially usefull for matching different UI states. `:` selector allows to match pseudo-classes like `:hover`, `:visited` and others. `::` selector matches pseudo-elements like `::before`, `::first-line` and others.
+
+*Example*: `button:hover` selector will match all buttons in a hovered state.
+
+
+## Data attributes
+
+[Data attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) is a type of HTML tag attributes which is especially usefull for defining actions in PostHog. Using them you can easily 'put' different events into one bucket. Data attrbutes are any attributes following the name convention `data-*` where `*` can be anything. 
+
+> Good thing about data attributes that they can be set on any HTML tag.
+
+Let's take a look at the example:
+
+```html
+   <button
+    type="submit"
+    data-action="newsletter__subscribe"
+    >
+       Subscribe
+   </button>
+```
+Here we defined data atribute `data-action` with a value `newsletter__subscribe`. As you see, we are in a full control of these attributes, meaning we can manage how narrow or wide our selectors will be. 
+
+Then in PostHog we can accurately group all events around newsletter subscription with a simple CSS selector:
+
+```css
+   [data-action='newsletter__subscribe']
+```
+
+If you want to selects only buttons, then you can do the following:
+
+```css
+   button[data-action='newsletter__subscribe']
+```
+
+Data attributes is a very flexible approach allowing to group together different events which otherwise would be really difficult to group. But with any powerfull tool, you should spend some time understanding what is the best way to apply it.
```

Comment:
```suggestion
Data attributes are very flexible, enabling you to group together different events which otherwise would be really difficult to group. But, as with any powerful tool, you need to spend some time to understanding the best way to use it.
```